controllers/tokenupload_controller.go

```diff
@@ -204,6 +176,8 @@ func logError(ctx context.Context, secret corev1.Secret, err error, r *TokenUplo
 
 }
 
+// Contract: having at only one event if upload failed and no events if uploaded.
+// For this need to delete the event every attempt
 func tryDeleteEvent(ctx context.Context, secretName string, ns string, r *TokenUploadReconciler, lg logr.Logger) {
```

Comment:
Why passing the reconciler as a parameter? Could this be a member method of the reconciler instead?